qt/applications/workbench/workbench/plotting/figureinteraction.py

```diff
@@ -136,6 +136,15 @@ def on_scroll(self, event):
             and len(event.inaxes.lines) == 0
         ):
             return
+
+        """
+        Disable the legend's draggability when the scroll event is
+        enabled to prevent the legend from unintentionally moving,
+        which is the default behavior in Matplotlib.
+        """
+        legend = event.inaxes.axes.get_legend()
+        legend_set_draggable(legend, False)
+
```

Comment:
This is a good start. I would suggest having an if statement to check whether the legend is already set to draggable or not. Because if the legend is not draggable already, we do not want to accidentally set it to true.

```suggestion
    if legend.get_draggable() and legend.contains(event):
        legend_set_draggable(legend, False)
        legend_set_draggable(legend, True)
```